livekit-plugins/livekit-plugins-deepgram/livekit/plugins/deepgram/stt.py

```diff
@@ -302,15 +304,59 @@ async def recv_task():
 
         await asyncio.gather(send_task(), recv_task(), keepalive_task())
 
+    def _commit_all_final_events(self) -> None:
+        if not self._speaking:
+            logging.warning(
+                "trying to commit final events without being in the speaking state"
+            )
+            return
+
+        if len(self._final_events) == 0:
+            logging.warning("received end of speech without any final transcription")
```

Comment:
@theomonnom I removed the return in this case so that a START_OF_SPEECH event is matched with an END_OF_SPEECH event even if it's likely a bug if we hit this condition